crates/ruff_python_parser/src/parser/expression.rs

```diff
@@ -193,12 +193,17 @@ impl<'src> Parser<'src> {
         self.parse_expression_with_precedence(Precedence::Initial)
     }
 
-    /// Binding powers of operators for a Pratt parser.
+    /// Returns the binding power of the current token for a Pratt parser.
     ///
-    /// See <https://matklad.github.io/2020/04/13/simple-but-powerful-pratt-parsing.html>
-    fn current_op(&mut self) -> (Precedence, TokenKind, Associativity) {
-        const NOT_AN_OP: (Precedence, TokenKind, Associativity) =
+    /// This includes the precedence and associativity of the current token.
+    /// If the current token is not an operator, it returns [`Precedence::Unknown`],
+    /// [`TokenKind::Unknown`], and [`Associativity::Left`] respectively.
+    ///
+    /// See: <https://matklad.github.io/2020/04/13/simple-but-powerful-pratt-parsing.html>
+    fn current_binding_power(&mut self) -> (Precedence, TokenKind, Associativity) {
```

Comment:
Nit: Unrelated to this PR:

I think it would be nice if we could refactor this a bit.

a) We should return an Option and return `None` if the current token is not an operator. That makes the error handling in `parse_expression_with_precedence` more explicit and allows us to remove `Precedence::Unknown` (I was worried that we call into `bump` when the operator is `Unkown` but I don't think that happens because `Unknown` always has a too low precedence)
b) The method is only used when parsing binary expressions. Could we return a `BinOp` instead and make `associativity` a method on `BinOp` (because all other operations are left associative).

Reply:
For (b), it returns an _infix_ operator which can be binary, boolean (`or` or `not) or bitwise. I think we can introduce a new `InfixOp` which could have `associativity` and `as_token` method.